### PR TITLE
Exposed permission configuration argument of local adapter

### PR DESCRIPTION
**Pull request by andreas-glaser**
"Exposed permission configuration argument of local adapter"

### CODE REVIEW

**Comment by bytehead** [comment]
Thanks for your contribution! I will check this asap.

File: DependencyInjection/Factory/Adapter/LocalFactory.php [hunk 1]
```diff
@@ -34,6 +35,25 @@ public function addConfiguration(NodeDefinition $node)
                 ->scalarNode('directory')->isRequired()->end()
                 ->scalarNode('writeFlags')->defaultValue(LOCK_EX)->end()
                 ->scalarNode('linkHandling')->defaultValue(Local::DISALLOW_LINKS)->end()
+                ->arrayNode('permissions')
+                    ->addDefaultsIfNotSet()
+                    ->children()
+                        ->arrayNode('file')
+                            ->addDefaultsIfNotSet()
+                            ->children()
+                                ->scalarNode('public')->defaultValue(0744)->end()
```
Comment (bytehead) [inline DependencyInjection/Factory/Adapter/LocalFactory.php]: Is there any reason for 744 instead of 644?

Comment (andreas-glaser) [inline DependencyInjection/Factory/Adapter/LocalFactory.php]: No not really i just followed flysystem's documentation at http://flysystem.thephpleague.com/adapter/local/

But looking at the current code base it is more restrictive.
https://github.com/thephpleague/flysystem/blob/master/src/Adapter/Local.php

I'll change that! Thanks

Comment (bytehead) [inline DependencyInjection/Factory/Adapter/LocalFactory.php]: Cool, thanks anyway!

**Comment by bytehead** [comment]
[File and directory permission settings [added in 1.0.14]](http://flysystem.thephpleague.com/adapter/local/).
I guess we have to increase the version constraint for  `league/flysystem` - don't we?

**Comment by andreas-glaser** [comment]
Yes good point it should probably be set to `"league/flysystem": "~1.0.14" `

**Comment by bytehead** [comment]
But then you cannot update it automatically to `> 1.0.*`...

> The ~ operator is best explained by example: ~1.2 is equivalent to >=1.2 <2.0.0, while ~1.2.3 is equivalent to >=1.2.3 <1.3.0

**Comment by bytehead** [comment]
Anyway, composer loads the newest `1.0.*` release of `league/flysystem`, so it shouldn't be a problem. Except someone really needs `league/flysystem < 1.0.14`.

**Comment by andreas-glaser** [comment]
Ok i personally would set it to `"league/flysystem": "^1.0.14"`. If anyone needs < 1.0.14 then they can stick to version 1.3.0 of you bundle.

**Comment by bytehead** [comment]
That's a good solution, thanks! I'll fix it asap.

**Pull request merged**